include/swift/AST/AutoDiff.h

```diff
@@ -330,6 +330,31 @@ enum class AutoDiffAssociatedVectorSpaceKind : unsigned {
   Tangent = 0, Cotangent = 1
 };
 
+/// The kind of ABI used to represent a differentiable function.
+enum class DifferentiabilityRepresentationKind : unsigned {
+  /// The function is linear and is represented as a bundle of the original
+  /// function and its transpose. Its differential is the function itself. Its
+  /// pullback is its transpose.
+  ///
+  /// For original function `(T...) -> U`, there are a few typing invariants:
+  /// 1. T = T.TangentVector = T.CotangentVector
+  /// 2. U = U.TangentVector = U.CotangentVector
+  ///
+  /// |----------------------|
+  /// | Original | Transpose |
+  /// |----------------------|
+  Linear,
+
+  /// The function is represented as a bundle of the original function and
+  /// JVP functions at every order. JVP functions must be thin.
```

Comment:
Don't we have thick JVP/VJPs now?

Are the JVPs here Linear bundles? If so, document that too.

Reply:
The idea is to require JVPs always share the closure context of the original function, and require `autodiff_function`'s operands to be thin. Any formation of a thick `@differentiable` function needs to be formed by `partial_apply` of a `@differentiable @convention(thin)` function.